product_pricelist_brand_required/models/product_pricelist_item.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2025 Angel Rivas <angel.rivas@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class PricelistItem(models.Model):
+    _inherit = "product.pricelist.item"
+
+    product_brand_ids = fields.Many2many("product.brand", string="Brands")
+
+    def _is_applicable_for(self, product, qty_in_product_uom):
+        self.ensure_one()
+        product.ensure_one()
+
+        if (
```

Comment:
change for:
```
        res = False
        if (
            not self.product_brand_ids
            or product.product_brand_id in self.product_brand_ids
        ):
           res = super()._is_applicable_for(product, qty_in_product_uom)
       return res
```

Reply:
Returns in a function:

- There should never be more than one return.
- It must always come at the end of the function.
- Using more than one return in a function makes it difficult to read and can cause errors.